### DISCUSSION TITLE
Multiple replicas

### DISCUSSION THREAD

**Discussion by begemotik** [outline/outline#4080] [Self Hosting] [answered]
We are running outline in Kubernetes and considering running multiple replicas of Outline container for High Availability, though didn't find any information in documentation about whether it's possible.

The question is quite simple as it is. May we? :)

**Comment by tommoor** [comment, marked as answer]
It's possible but the collaboration service must only be run once [SERVICES.md](https://github.com/outline/outline/blob/main/docs/SERVICES.md). I wouldn't recommend it as being worth the hassle.

**Reply by begemotik** [reply]
Though I think that would be nice to have that possibility. So, to conclude, we need to have 2 separate containers with different launching commands:

1. All the services except "collaboration", add env var `COLLABORATION_URL=wss://docs-collaboration.example.com`. This container can be scaled to multiple replicas.
```
yarn start --services=web,worker,admin,websockets
```
2. "Collaboration" service. This one can't be scaled.
```
yarn start --services=collaboration
```

Is that right?

**Reply by tommoor** [reply]
Correct, although it's unwise to include "admin" in the list of services outside of development.

**Reply by almereyda** [reply]
Eventually, in Kubernetes one could run a separate `admin` instance and only expose it as a ClusterIP service without Ingress, similar to what Traefik does with its API and dashboard. One then accesses it through an independent Identity-aware Access Proxy (Pomerium, OAuth2-Proxy) or via local port forwards.

**Comment by almereyda** [comment]
@begemotik Can you share your Helm chart?

**Reply by begemotik** [reply]
unfortunatelly, no. We use our internal helm library

**Reply by almereyda** [reply]
Thank you. Let's see what the community comes up with, then.

If I may ask, could you see with your organisation if you could open source your Outline-related work, eventually under the BSL or a similar license?
